springwolf-plugins/springwolf-cloud-stream-plugin/src/test/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/CloudStreamFunctionChannelsScannerTest.java

```diff
@@ -0,0 +1,315 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.channels;
+
+import com.asyncapi.v2.binding.ChannelBinding;
+import com.asyncapi.v2.binding.OperationBinding;
+import com.asyncapi.v2.model.channel.ChannelItem;
+import com.asyncapi.v2.model.channel.operation.Operation;
+import com.asyncapi.v2.model.info.Info;
+import com.asyncapi.v2.model.server.Server;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import io.github.stavshamir.springwolf.asyncapi.scanners.beans.DefaultBeanMethodsScanner;
+import io.github.stavshamir.springwolf.asyncapi.scanners.classes.ConfigurationClassScanner;
+import io.github.stavshamir.springwolf.asyncapi.types.channel.operation.message.Message;
+import io.github.stavshamir.springwolf.asyncapi.types.channel.operation.message.PayloadReference;
+import io.github.stavshamir.springwolf.asyncapi.types.channel.operation.message.header.AsyncHeaders;
+import io.github.stavshamir.springwolf.asyncapi.types.channel.operation.message.header.HeaderReference;
+import io.github.stavshamir.springwolf.configuration.AsyncApiDocket;
+import io.github.stavshamir.springwolf.schemas.DefaultSchemasService;
+import org.apache.kafka.streams.kstream.KStream;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.stream.config.BindingProperties;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {
+        ConfigurationClassScanner.class,
+        DefaultBeanMethodsScanner.class,
+        DefaultSchemasService.class,
+        CloudStreamFunctionChannelsScanner.class
+})
+@Import(CloudStreamFunctionChannelsScannerTest.Configuration.class)
+public class CloudStreamFunctionChannelsScannerTest {
+
+    @MockBean
+    private BindingServiceProperties bindingServiceProperties;
+
+    @Autowired
+    private CloudStreamFunctionChannelsScanner scanner;
+
+    @Test
+    public void noBindings() {
```

Comment:
Test methods would benefit from the `test` method name prefix as in the other test to separate them from beans, etc.

Reply:
Will change